src/arch/x64/mod.rs

```diff
@@ -26,4 +26,6 @@ pub mod context;
 
 pub mod memory;
 
-global_asm!(include_str!("routines.asm"));
\ No newline at end of file
+pub mod rand;

```

Comment:
I think `rand` should be in `arch/devices/rand.rs`, instead of a top-level arch module.